### PR TITLE
[MappingApp] python-mapper-factory

### PR DESCRIPTION
**Pull request by philbucher**
during a discussion this week @RiccardoRossi asked @rubenzorrilla and me to unify how the mapping is used in Kratos.
Right now we have mapping implemented in many different places, but the plan is to eventually implement them in the MappingApp, such that they also work in MPI

For now we thought it would be a good idea if we use a "common" place where to construct the mappers, and we agreed that it can only be done in python for now 
=> hence I introduced the `python_mapper_factory`, where we can also register the mappers from the others apps/the core without introducing dependencies.
This way I can eventually replace the mappers under the hood without the users having to change stuff

The interfaces of the mappers are slightly different, thats why I could not integrate the other mappers right away.
I have been discussing with each of you already that we should unify the interface, for which [`mapper.h`](https://github.com/KratosMultiphysics/Kratos/blob/master/applications/MappingApplication/custom_mappers/mapper.h) is very suitable. This is still (!) the interface that was decided on during my master thesis, it works nicely

@rubenzorrilla already agreed to update the interface of the mapper in the FSI-App, I can also speak to the others in person and discuss it

### CODE REVIEW

File: applications/MappingApplication/python_scripts/python_mapper_factory.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    import KratosMultiphysics.FSIApplication as KratosFSI
+    # return mapper from FSIApp
+    raise NotImplementedError
+
+def _CreateVertexMorphingMapper(model_part_origin, model_part_destination, mapper_settings):
+    import KratosMultiphysics.ShapeOptimizationApplication as KratosShapeOpt
+    # return mapper from ShapeOptApp
+    raise NotImplementedError
+
+def _CreateEmpireMortarMapper(model_part_origin, model_part_destination, mapper_settings):
```
Comment (philbucher) [inline applications/MappingApplication/python_scripts/python_mapper_factory.py:25]: @emiroglu-statik this is what we were discussing last week, now it is possible :)
@andreas-apostolatos I think this is also interesting for you. Having the empire-mappers easily available would be very useful to compare them, esp the IGA one

**Review by loumalouomega — COMMENTED** [review]
This is OK from my side

**Comment by rubenzorrilla** [comment]
OK from my side as well. I'll update the FSI app as soon as this is merged.

**Comment by armingeiser** [comment]
Fine for me. I think the VM-mapper interface is slightly different but ii guess it could be adapted in a next step.

**Comment by philbucher** [comment]
can someone approve please then we can start with the migration/unification
Thx

**Comment by philbucher** [comment]
none?

**Review by loumalouomega — APPROVED** [review]
Feel free

**Comment by philbucher** [comment]
thx

**Pull request merged**